src/settings.rs

```diff
@@ -170,6 +170,7 @@ impl Settings {
             pyproject,
             project_root,
         };
+        settings.select(config.extend_select);
         settings.ignore(&config.ignore);
```

Comment:
I feel like the semantics here are off (but that they were off before with `settings.ignore(&config.ignore)` -- not any fault of your change). We should probably avoid calling `settings.ignore(&config.ignore)`, since if the user provides a `--ignore` on the command-line, we'd want to respect that _instead_ (not in addition). Same for `extend_select`.

But this can be changed separately, what you have here is an improvement in overall behavior.